lib/puppet_x/puppetlabs/strings/yard/handlers/host_class_handler.rb

```diff
@@ -8,6 +8,21 @@ class PuppetX::PuppetLabs::Strings::YARD::Handlers::HostClassHandler < PuppetX::
         param_tuple << ( a[1].nil? ? nil : a[1].source )
       end
     end
+    tp = Puppet::Pops::Types::TypeParser.new
+    param_type_info = []
+    # Ensure that this pop object has parameters
+    if statement.pops_obj.respond_to? :parameters
+      param_type_info = {}
+      statement.pops_obj.parameters.each do |pop_param|
+        begin
+          param_type_info[pop_param.name] =  tp.interpret_any(pop_param.type_expr)
+        rescue Puppet::ParseError
```

Comment:
I think a nil type_expr should be handled differently from a broken type_expr - when it is nil, it should default to Any. But if there is an error the documentation should say something else e.g if given `Integer[herp, derp]` - then the result should not be Any - maybe emit something like "TypeError - <message>"